lib/logger/lib/logger/utils.ex

```diff
@@ -244,14 +244,14 @@ defmodule Logger.Utils do
         true  -> :calendar.now_to_universal_time(now)
         false -> :calendar.now_to_local_time(now)
       end
-    {date, {hours, minutes, seconds, div(micro, 1000)}}
+    {date, {hours, minutes, seconds, micro}}
   end
 
   @doc """
   Formats time as chardata.
   """
   def format_time({hh, mi, ss, ms}) do
-    [pad2(hh), ?:, pad2(mi), ?:, pad2(ss), ?., pad3(ms)]
```

Comment:
In order to not change the output currently logged, let's make this be:

```elixir
[pad2(hh), ?:, pad2(mi), ?:, pad2(ss), ?., ms |> div(1000) |> pad3()]
```

If you want to log with higher precision, then you can always call your own `format_time` function.

This way you get higher precision in the backend but remove the precision only when formatting, as you mentioned.